libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/transition/MapboxNavigationCameraTransition.kt

```diff
@@ -304,7 +304,10 @@ class MapboxNavigationCameraTransition(
         val currentZL = currentMapCameraState.zoom
         val targetCenter = cameraOptions.center
         val targetZoom = cameraOptions.zoom
-        val animators = pluginImpl.cameraAnimationsFactory.getFlyTo(cameraOptions)
+        val animators = pluginImpl.cameraAnimationsFactory.getFlyTo(
+            cameraOptions = cameraOptions,
+            owner = NAVIGATION_CAMERA_OWNER
```

Comment:
If the owner is not set, the generated here fly-to camera transition is cancelable by code like https://github.com/mapbox/mapbox-navigation-android/blob/3b4e399e3c3fdbce2471a6c923ce9d533f3bab38/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/lifecycle/NavigationScaleGestureHandler.kt#L145-L147 which cancels Navigation Camera if any external (so other than Navigation Camera itself and gestures) transition is executed to avoid competition of updates. In the case of lacking the owner, the Navigation Camera transition can effectively cancel itself, so the transitions never execute.

This wasn't reproducible when animations are run on the main thread because then the init -> start of the transition is synchronous so the first tick is effectively ignored because the state wasn't set yet (perhaps there's another small issue somewhere in there connected with relying on callback order, I'm not sure). However, when we move the animators to run on a worker thread, the transitions are initialized and only later at some point in time started, so the cancellation was effective.